docs/premium-features/custom-types-and-actions.md

```diff
@@ -0,0 +1,259 @@
+---
+title: Custom Types & Actions
+sidebar_label: Custom Types & Actions
+slug: /custom-types-and-actions
+pagination_next: api/index
+---
+
+# Custom Types & Actions
+
+Custom Types & Actions is an enhanced feature that gives you full control over your Data Transfer Objects (DTOs) and APIs. It allows you to automatically generate high-quality, efficient code for both the default CRUD operations and your custom APIs that utilize custom DTOs, all within your REST and/or GraphQL APIs.
+
+Custom Types & Actions let you define custom data structures (Types), create API endpoints (Actions), and manage the relationships between them, all within the Amplication UI. These changes are automatically translated into production-ready code in your repository, ensuring the best quality and efficiency.
+
+:::note
+The Custom Types & Actions feature is available for Amplication's **Enterprise** plan, including trial plans.
+:::
+
+## What Are Types And Actions?
+
+In Amplication, Types and Actions are fundamental concepts that define the data structures and API endpoints of your application.
+
+### Types
+
+Types in Amplication refer to the data structures used to define the shape and format of your application's data. There are two main categories of types:
+
+1. **Data Transfer Objects (DTOs)**: DTOs are custom data structures that define the input and output data for your API endpoints (Actions). They encapsulate the data that is transferred between the client and the server or between different services within your application.
+
+2. **Primitives**: Primitives are the basic data types supported by Amplication, such as String, Integer, Boolean, Date, and JSON. These primitives can be used as fields within your DTOs or as standalone types for your Actions.
+
+### Actions
+
+Actions in Amplication represent the API endpoints that define the operations available in your application. Each Action specifies the input, output, and other configurations specific to the API type (REST or GraphQL).
+
+For [REST API](/api/#rest-apis), an Action defines:
+
+- HTTP method (GET, POST, PUT, DELETE, etc.)
+- Path (e.g., `/users` or `/products/:id`)
+- Input parameters (query, path, or request body)
+- Output data structure (response body)
+
+For [GraphQL API](/api/#graphql-apis), an Action defines:
+
+- Operation type (Query or Mutation)
+- Input type (arguments)
+- Output type (return type)
+
+Actions can be of two types:
+
+1. **Default CRUD Actions**: Amplication automatically generates default CRUD (Create, Read, Update, Delete) Actions based on your application's entities. These Actions provide standard API endpoints for interacting with your data.
+
```

Comment:
Consider if we want to add information about the default *relation* CRUD operations (e.g. get all books for author, get author for book, connect/disconnect book to author) 
@PazYanoverr - it's not mandatory, for your consideration

Reply:
Sure, @dericksozo - Let's add it.